### ISSUE TITLE
XML extractor fails in directories containing spaces 

### ISSUE THREAD

**Issue by Marcono1234** [github/codeql#8778]
### Version
CodeQL CLI version 2.8.5
OS: Windows 10

### Description
It looks like the XML extractor fails when executed in a directory whose name contains spaces:
```
Initializing database at ...\with spaces\codeql-db.
Running build command: [javac, Test.java]
Finalizing database at ...\with spaces\codeql-db.
[2022-04-19 00:21:35] [build-stderr] Scanning for files in ...\with spaces...
[2022-04-19 00:21:35] [build-stderr] ...\with spaces\codeql-db: Indexing files in in ...\with spaces...
[2022-04-19 00:21:35] [build-stderr] Running command in ...\with spaces: [...\codeql\xml\tools\index-files.cmd, ...\with spaces\codeql-db\working\files-to-index11477293421044941804.list]
[2022-04-19 00:21:36] [build-stdout] [2022-04-19 00:21:36] [build-stderr] Option only takes one value: [--fileList value] : A file containing a list of absolute file paths to index, regardless of exclusions
[2022-04-19 00:21:36] [build-stdout] [2022-04-19 00:21:36] [build-stderr] Option only takes one value: [--sourceArchiveDir value] : Directory to place archived copy of files into
[2022-04-19 00:21:36] [build-stdout] [2022-04-19 00:21:36] [ERROR] Spawned process exited abnormally (code 1; tried to run: [...\codeql\tools\win64\runner.exe, cmd.exe, /C, type, NUL, &&, ...\codeql\xml\tools\index-files.cmd, ...\with spaces\codeql-db\working\files-to-index11477293421044941804.list])
[2022-04-19 00:21:36] [build-stderr] A fatal error occurred: Exit status 1 from command: [...\codeql\tools\win64\runner.exe, cmd.exe, /C, type, NUL, &&, ...\codeql\xml\tools\index-files.cmd, ...\with spaces\codeql-db\working\files-to-index11477293421044941804.list]
[2022-04-19 00:21:36] [ERROR] Spawned process exited abnormally (code 2; tried to run: [...\codeql\tools\win64\runner.exe, cmd.exe, /C, type, NUL, &&, ...\codeql\java\tools\pre-finalize.cmd])
A fatal error occurred: Exit status 2 from command: [...\codeql\tools\win64\runner.exe, cmd.exe, /C, type, NUL, &&, ...\codeql\java\tools\pre-finalize.cmd]
```

Maybe other extractors are affected as well.

### Reproduction steps
1. Create a new directory called `with spaces`
2. In that directory, create a file called `Test.java` with the following content
    ```java
    class Test {}
    ```
3. Create a file called `test.xml` (can be empty)
4. Run the following command
    ```
    codeql database create --overwrite --language=java "--command=javac Test.java" codeql-db
    ```
    :x: It fails with the error message shown above

**Comment by aeisenberg** [comment]
Thanks for raising this issue. I am transferring this to the team responsible for maintaining the java extractor.

**Comment by smowton** [comment]
Fix committed and will go out in (I think) v2.9.1.
